source/includes/read/cursors.kt

```diff
@@ -0,0 +1,65 @@
+package org.example
+import com.mongodb.ConnectionString
+import com.mongodb.CursorType
+import com.mongodb.MongoClientSettings
+import com.mongodb.client.model.Filters.eq
+import com.mongodb.kotlin.client.MongoClient
+import org.bson.codecs.pojo.annotations.BsonId
+import org.bson.types.ObjectId
+
+// start-data-class
+data class Restaurant(
+    @BsonId
+    val id: ObjectId,
+    val name: String
+)
+// end-data-class
+
+fun main() {
+    val uri = "<connection string URI>"
+
+    val settings = MongoClientSettings.builder()
+        .applyConnectionString(ConnectionString(uri))
+        .retryWrites(true)
+        .build()
+
+    val mongoClient = MongoClient.create(settings)
+    val database = mongoClient.getDatabase("sample_restaurants")
+    val collection = database.getCollection<Restaurant>("restaurants")
+
+    // start-cursor-iterate
+   val results = collection.find()
+
+   results.forEach { result ->
+       println(results)
+   }
+    // end-cursor-iterate
+
+    // start-cursor-next
+   val results = collection.find<Restaurant>(eq(Restaurant::name.name, "Dunkin' Donuts"))
+
+   println(results.cursor().next())
+    // end-cursor-next
+
+    // start-cursor-list
+    val results = collection.find<Restaurant>(eq(Restaurant::name.name, "Dunkin' Donuts"))
+    val resultsList = results.toList()
+
+    for (result in resultsList) {
+        println(result)
+    }
+    // end-cursor-list
+
+    // start-cursor-close
+    val results = collection.find<Restaurant>()
+
+    // Handle your results here
+
+    results.cursor().close()
+    // end-cursor-close
```

Comment:
Closing a cursor explicitly (well, kind of, as `close` is still not explicit in the source code) should be done via the [`use`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html) function:

```suggestion
    // start-cursor-close
    collection.find<Restaurant>().cursor().use { cursor ->
        // Handle your results here
        cursor.close()
    }
    // end-cursor-close
```

Reply:
I'll make suggestions for the changed code separately. One of them is to remove `close()` from the body of function passed to `use` methods, because `use` calls `close` on its own. It was my typo to leave `cursor.close()` in the proposed code.